tests/helpers/auth-helper.test.ts

```diff
@@ -0,0 +1,136 @@
+    });
+
+    test('generatePasswordHash should generate a valid hash', async () => {
+        expect.assertions(1);
+        const testPasswordHash = await getPasswordHash();
+        const passwordCheck =
+            await authHelper.checkPasswordHash(testPassword, testPasswordHash);
+        expect(passwordCheck).toBe(true);
+    });
+
+    test('generateJWT should return a string', () => {
+        const payloadData = {
+            user: 'Test',
+        };
+        const authToken = authHelper.generateJWT(payloadData);
+        expect(typeof authToken).toBe('string');
+    });
+
+    test('decodeJWT should return the correct payload', () => {
+        const payloadData = {
+            user: 'Test',
+        };
+        const payload = authHelper.generatePayload(payloadData);
+        const authToken = authHelper.generateJWT(payloadData);
+        const decodedPayload = authHelper.decodeJWT(authToken);
+        expect(decodedPayload).toEqual(payload);
+        expect(decodedPayload.data).toEqual(payloadData);
+    });
+
+    test('JWT dates create a valid date object', () => {
+        const payload = {
+            user: 'Test',
+        };
+        const authToken = authHelper.generateJWT(payload);
+        const decodedPayload = authHelper.decodeJWT(authToken);
+        const payloadDate = new Date(decodedPayload.expires);
+        expect(payloadDate).toBeInstanceOf(Date);
+    });
+
+    test('JWT expiry date should be in the future', () => {
+        const payload = {
+            user: 'Test',
+        };
+        const authToken = authHelper.generateJWT(payload);
+        const decodedPayload = authHelper.decodeJWT(authToken);
+        const payloadIssuedAtDate = decodedPayload.iat;
+
+        // Precise to 1 second TODO: Test this better
+        const issuedDate = Math.round(payloadIssuedAtDate / 1000);
+        const nowDate = Math.round(Date.now() / 1000);
+        expect(issuedDate).toBe(nowDate);
+    });
+
+    test('JWT issued at date should be now', () => {
+        const payload = {
+            user: 'Test',
+        };
+        const authToken = authHelper.generateJWT(payload);
+        const decodedPayload = authHelper.decodeJWT(authToken);
+        const payloadIssuedAt = decodedPayload.iat;
+
+        // Precise to 1 second TODO: Test this better
+        const payloadIAT = Math.round(payloadIssuedAt / 1000);
+        const expectedIAT = Math.round(Date.now() / 1000);
+        expect(payloadIAT).toBe(expectedIAT);
+    });
+
+    test('JWT should expire in JWT_EXPIRES_IN_DAYS days', () => {
+        const payload = {
+            user: 'Test',
+        };
+        const authToken = authHelper.generateJWT(payload);
+        const decodedPayload = authHelper.decodeJWT(authToken);
+        const payloadIssuedAtDate = decodedPayload.iat;
+        const payloadExpiryDate = decodedPayload.exp;
+        const calculatedDays = authHelper.calculateDates();
+
+        // Precise to 1 second TODO: Test this better
```

Comment:
@JeroenVanSteijn do you perhaps have some thoughts on how this van be done cleaner?

Reply:
Could you point out to me which exact feature in the code you're trying to test?

Reply:
It's about the generation of timestamps, for "Issued At" and "Expires At". Code:

```js
const calculateDates = issuedAtParam => {
    const date = new Date();
    const issuedAt = issuedAtParam || date.setDate(date.getDate());
    const issuedAtDate = new Date(issuedAt);
    const expiryDate = date.setDate(issuedAtDate.getDate() + jwtExpiresInDays);
    const dates = {
        iat: issuedAt,
        exp: expiryDate,
    };
    return dates;
};

const generatePayload = (data, issuedAt) => {
    const dates = calculateDates(issuedAt);
    const payload = {
        ...dates,
        data,
    };
    return payload;
};

const generateJWT = (data, issuedAt) => {
    const payload = generatePayload(data, issuedAt);
    const token = jwt.encode(payload, JWT_SECRET);
    return token;
};
```